games-board/stockfish/stockfish-sf_8.recipe

```diff
@@ -0,0 +1,72 @@
+SUMMARY="A chess engine claimed to be strongest in the world"
+DESCRIPTION="Stockfish is a free UCI chess engine derived from Glaurung 2.1. It \
+is not a complete chess program and requires some UCI-compatible GUI (e.g. \
+XBoard with PolyGlot, eboard, Arena, Sigma Chess, Shredder, ChessPartner or \
+Fritz) in order to be used comfortably. Read the documentation for your GUI of \
+choice for information about how to use Stockfish with it."
+HOMEPAGE="https://www.stockfishchess.org/"
+COPYRIGHT="2004-2008 Tord Romstad (Glaurung author)
+	2008-2017 Marco Costalba, Joona Kiiski, Gary Linscott, Tord Romstad"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/official-stockfish/Stockfish/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="e2b77d801d8837c22c2391b8cae08622c1434a2bb5947ddb08b081e1ad35c76c"
+SOURCE_DIR="Stockfish-sf_8"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64 ?arm ?ppc"
+SECONDARY_ARCHITECTURES="!x86_gcc2 x86"
+
+if [ "$targetArchitecture" != x86_gcc2 ]; then
+	if [ "$targetArchitecture" == arm ]; then
+		commandBinDir=$binDir
+		commandArch=armv7
+	elif [ "$targetArchitecture" == ppc ]; then
+		commandBinDir=$binDir
+		commandArch=ppc-32
+	elif [ "$targetArchitecture" == x86 ]; then
+		commandBinDir=$prefix/bin
+		commandArch=x86-32
+	elif [ "$targetArchitecture" == x86_64 ]; then
+		commandBinDir=$binDir
+		commandArch=x86-64
+	fi
+else
+	commandBinDir=$prefix/bin
+	commandArch=x86-32
+fi
```

Comment:
So you only need to set commandBinDir to $prefix/bin if both of these conditions are met:
* the software fails to build on x86_gcc2 primary arch
* haikuporter is building for the x86 secondary arch.

So I guess you should use the same block as you did for haikuports/sci-libs/primesieve/primesieve-5.7.3.
Alternatively, you can use the following variant, which we have on many recipes, except that they also define commandSuffix. (BTW, I guess that also using commandSuffix would be usefull in the future if we create a x86_64 hybrid, i.e. with 64 bits for the primary arch and 32 for a x86 secondary arch.)
```
commandBinDir=$binDir
if [ "$targetArchitecture" = x86_gcc2 ]; then
	commandBinDir=$prefix/bin
fi
```

Regarding the commandArch, I think you shoud use something like:
```
case "$effectiveTargetArchitecture" in
	x86) commandArch=x86-32;;
	x86_64) commandArch=x86-64;;
	arm) commandArch=armv7;;
	ppc) commandArch=ppc-32;;
esac
```
But I don't know if it's worth adding ppc and arm right now, since our nightly builds for these architectures are not currently disabled (due to missing devs or bootstrap issue, I don't know exactly).
So let's see what the other mentors say about that, i.e. if you can keep arm and ppc in this recipe.
Since you have investigated about the ARCH parameter to pass for them, they might agree with keeping them here, although it won't be possible to build the recipe for arm or ppc right now ;-)